utils/sub8_ros_tools/sub8_ros_tools/geometry_helpers.py

```diff
@@ -70,3 +73,36 @@ def compose_transformation(R, t):
     transformation[3, :3] = t
     transformation[3, 3] = 1.0
     return transformation
+
+
+def quat_to_rotvec(q):
+    ''' Turn a quaternion into a axis rotation vector [x,y,z] '''
+    if q[3] < 0:
+        q = -q
+    q = transformations.unit_vector(q)
+    angle = math.acos(q[3])*2
+    axis = normalize(q[0:3])
```

Comment:
This will result in a divide-by-zero if this function is given the identity quaternion. This function appears to have been copied from `uf_common`, but it depends on behavior that the `normalize` here doesn't have.